File: docs/guides/examples/cpp/src/create_bucket.cc
#include <reduct/client.h>
#include <iostream>
#include <cassert>

using reduct::IBucket;
using reduct::IClient;
using reduct::Error;

int main() {
  // Create a client with the server URL
  auto client = IClient::Build("http://127.0.0.1:8383", {
    .api_token = "my-api-token"
  });

  // Create a bucket with the name "my-bucket" and a FIFO quota of 1GB
  IBucket::Settings settings;
  settings.quota_type = IBucket::QuotaType::kFifo;
   settings.quota_size = 1'000'000'000;
  auto [bucket, create_err] = client->GetOrCreateBucket("my-bucket", settings);

  assert(create_err == Error::kOk);
  assert(bucket->GetInfo().result.name == "my-bucket2");
}

File: docs/guides/examples/mjs/src/create_bucket.mjs
import {Client, QuotaType} from "reduct-js";
import assert from "node:assert";

// Create a new client with the server URL and an API token
const client = new Client("http://127.0.0.1:8383", {apiToken: "my-api-token"});

// Create a bucket with the name "my-bucket" and a FIFO quota of 1GB
let settings = {
  quotaType: QuotaType.FIFO,
  quotaSize: 1000_000_000n,
}
const bucket = await client.getOrCreateBucket("bucket", settings);
assert(bucket.name === "bucket1");